akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -410,19 +432,32 @@ private[akka] class Shard(
     case RestartEntities(ids) => restartEntities(ids)
   }
 
-  private def receiveStartEntity(start: ShardRegion.StartEntity): Unit = {
-    val requester = sender()
-    log.debug("Got a request from [{}] to start entity [{}] in shard [{}]", requester, start.entityId, shardId)
-    touchLastMessageTimestamp(start.entityId)
+  // this could be because of a start message or due to a new message for the entity
+  // if it is a start entity then start entity ack is sent after it is created
+  private def startEntities(entities: Map[EntityId, Option[ActorRef]]): Unit = {
+    val alreadyStarted = entities.filterKeys(entity => entityIds(entity))
```

Comment:
When are they already started?

Reply:
When we receive a StartEntity but it's already been triggered by a message